sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2571,12 +2605,61 @@ __ESIMD_API simd<Tx, N> slm_atomic_update(simd<uint32_t, N> offsets,
 
 /// @} sycl_esimd_memory_slm
 
+namespace detail {
+
+/// USM pointer atomic.
+/// Supported platforms: DG2, PVC
+/// VISA instruction: lsc_atomic_<OP>.ugm
+///
+/// @tparam Op is operation type.
+/// @tparam T is element type.
+/// @tparam N is the number of channels (platform dependent).
+/// @tparam L1H is L1 cache hint.
+/// @tparam L3H is L3 cache hint.
+/// @param p is the base pointer.
+/// @param offsets is the zero-based offsets.
+/// @param pred is predicates.
+///
+template <__ESIMD_NS::atomic_op Op, typename T, int N,
+          cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
+          typename Toffset>
+__ESIMD_API std::enable_if_t<__ESIMD_DNS::get_num_args<Op>() == 0,
+                             __ESIMD_NS::simd<T, N>>
+atomic_update_impl(T *p, __ESIMD_NS::simd<Toffset, N> offsets,
+                   __ESIMD_NS::simd_mask<N> pred) {
+  static_assert(std::is_integral_v<Toffset>, "Unsupported offset type");
+  static_assert(sizeof(T) > 1, "Unsupported data type");
+  detail::check_lsc_vector_size<1>();
+  __ESIMD_DNS::check_atomic<Op, T, N, 0, /*IsLSC*/ true>();
+  detail::check_cache_hint<detail::cache_action::atomic, L1H, L3H>();
+  constexpr uint16_t _AddressScale = 1;
```

Comment:
nit: Using underscore is not required and it makes the code not following the clang coding std.
Let's use DS, VS, ImmOffset, etc

Reply:
Done in https://github.com/intel/llvm/pull/11833/commits/ffe19702c45355afd8b7ab17d2170cd783e9f670.